### PR TITLE
Yuval ui infra

### PR DESCRIPTION
**Pull request by yhaspel**
In review...

### CODE REVIEW

File: app/components/baseComponents/SendReceiveButton/SendReceiveButton.js [hunk 1]
```diff
@@ -0,0 +1,125 @@
+// @flow
+import * as React from 'react';
+import Colors from '../../../vars/colors';
```
Comment (IlyaVi) [inline app/components/baseComponents/SendReceiveButton/SendReceiveButton.js]: import { smColors, smFonts } from '../../vars';

File: app/components/baseComponents/SmButton/SmButton.js [hunk 2]
```diff
@@ -0,0 +1,121 @@
+// @flow
+import * as React from 'react';
```
Comment (IlyaVi) [inline app/components/baseComponents/SmButton/SmButton.js]: import React from 'react';

File: app/components/baseComponents/SmDropdown/SmDropdown.js [hunk 3]
```diff
@@ -0,0 +1,315 @@
...
+        height: 6,
+        width: 10,
+        transform: `rotate(${itemsHeight === ROW_HEIGHT ? '0' : '180'}deg)`,
+        transition: 'transform .2s linear',
+        zIndex: 999
+      }
+    };
+
+    const outerStyle = (isFirst: boolean) => {
+      if (!isFirst) {
```
Comment (IlyaVi) [inline app/components/baseComponents/SmDropdown/SmDropdown.js]: const outerStyle = (isFirst: boolean) => (!isFirst ? { borderTop: `1px solid ${Colors.borderGray}` } :  {});

File: app/components/baseComponents/SmDropdown/SmDropdown.js [hunk 4]
```diff
@@ -0,0 +1,315 @@
...
+      }
+    }, 0);
+  }
+
+  handleMouseEnter = (id: number | string) => {
+    this.setState({ hovered: id });
+  };
+
+  handleMouseLeave = () => {
+    this.setState({ hovered: undefined });
```
Comment (IlyaVi) [inline app/components/baseComponents/SmDropdown/SmDropdown.js]: change to -1

File: app/components/baseComponents/SmDropdown/SmDropdown.js [hunk 5]
```diff
@@ -0,0 +1,315 @@
...
+  };
+
+  handleHeaderEnter = () => {
```
Comment (IlyaVi) [inline app/components/baseComponents/SmDropdown/SmDropdown.js]: handleHeaderEnter = () => {
    const { disabled } = this.props;
    if (!disabled) {
      this.setState({ hoveredHeader: true });
    }
  };

File: app/components/baseComponents/SmDropdown/SmDropdown.js [hunk 6]
```diff
@@ -0,0 +1,315 @@
...
+
+  _getEntryByID = (id: string | number): ?DropdownEntry => {
+    const { data } = this.props;
+    return data.find((dropdownEntry: DropdownEntry) => dropdownEntry.id === id);
+  };
+
+  _getWidth = (): number | string => {
+    const { width, data } = this.props;
+    const LETTER_WIDTH = 10;
+    if (width === undefined) {
```
Comment (IlyaVi) [inline app/components/baseComponents/SmDropdown/SmDropdown.js]: !width

File: app/components/baseComponents/SmInput/SmInput.js [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+        onChange={this.handleChangeText}
+        className={styles.noOutline}
+        placeholder={placeholder || INPUT_PLACEHOLDER}
+        data-tid="noOutline"
+        style={{ ...inlineStyles.normal, ...additionalStyle() }}
+      />
+    );
+
+    return (
+      <div>
```
Comment (IlyaVi) [inline app/components/baseComponents/SmInput/SmInput.js]: remove div

File: app/vars/index.js [hunk 8]
```diff
@@ -1,2 +1,2 @@
 export { default as smColors } from './colors';
-export { default as routes } from './routes';
+export { default as routes } from '../routes';
```
Comment (IlyaVi) [inline app/vars/index.js]: remove

Comment (yhaspel) [reply app/vars/index.js]: Done

File: app/components/baseComponents/SmInput/SmInput.js (hunk 7 above)
Comment (yhaspel) [reply app/components/baseComponents/SmInput/SmInput.js]: done

File: app/components/baseComponents/SmDropdown/SmDropdown.js (hunk 6 above)
Comment (yhaspel) [reply app/components/baseComponents/SmDropdown/SmDropdown.js]: done

File: app/components/baseComponents/SmDropdown/SmDropdown.js (hunk 5 above)
Comment (yhaspel) [reply app/components/baseComponents/SmDropdown/SmDropdown.js]: done

File: app/components/baseComponents/SmDropdown/SmDropdown.js (hunk 4 above)
Comment (yhaspel) [reply app/components/baseComponents/SmDropdown/SmDropdown.js]: done

File: app/components/baseComponents/SmDropdown/SmDropdown.js (hunk 3 above)
Comment (yhaspel) [reply app/components/baseComponents/SmDropdown/SmDropdown.js]: ok

File: app/components/baseComponents/SmButton/SmButton.js (hunk 2 above)
Comment (yhaspel) [reply app/components/baseComponents/SmButton/SmButton.js]: ok

File: app/components/baseComponents/SendReceiveButton/SendReceiveButton.js (hunk 1 above)
Comment (yhaspel) [reply app/components/baseComponents/SendReceiveButton/SendReceiveButton.js]: done

File: app/vars/fonts.js [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+  fontSize: number,
+  fontWeight: 'bold' | 'normal' | '100' | '200' | '300' | '400' | '500' | '600' | '700' | '800' | '900' | string
+};
+
+const fontFamilies = {
+  openSans: 'sans-serif'
+};
+
+const Fonts: { [key: string]: Font } = {
+  font1: {
```
Comment (IlyaVi) [inline app/vars/fonts.js]: give fonts meaningful names

File: app/components/baseComponents/SendReceiveButton/SendReceiveButton.js [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+export default class SendReceiveButton extends React.Component<SendReceiveButtonProps, SendReceiveButtonState> {
+  state = {
+    hovered: false
+  };
+
+  render() {
+    const { disabled, onPress, title } = this.props;
+    const { hovered } = this.state;
+
+    const imageStyle = { height: 40, width: 40 };
```
Comment (IlyaVi) [inline app/components/baseComponents/SendReceiveButton/SendReceiveButton.js]: export to const

File: app/components/baseComponents/SendReceiveButton/SendReceiveButton.js [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+
+    const getImageSource = () => {
```
Comment (IlyaVi) [inline app/components/baseComponents/SendReceiveButton/SendReceiveButton.js]: extract to class method

File: app/components/baseComponents/SendReceiveButton/SendReceiveButton.js [hunk 12]
```diff
@@ -0,0 +1,115 @@
...
+      if (disabled) {
+        return title === 'Send coins' ? sendImageSourceDisabled : receiveImageSourceDisabled;
+      } else {
+        return title === 'Send coins' ? sendImageSource : receiveImageSource;
+      }
+    };
+
+    const sendReceiveButtonDisabledStyle = () => (disabled ? styles.disabledText : {});
```
Comment (IlyaVi) [inline app/components/baseComponents/SendReceiveButton/SendReceiveButton.js]: same

File: app/components/baseComponents/SendReceiveButton/SendReceiveButton.js [hunk 13]
```diff
@@ -0,0 +1,115 @@
...
+
+    const sendReceiveButtonElem = () => (
```
Comment (IlyaVi) [inline app/components/baseComponents/SendReceiveButton/SendReceiveButton.js]: same

File: app/components/baseComponents/SmRadioButtons/SmRadioButtons.js [hunk 14]
```diff
@@ -0,0 +1,167 @@
+// @flow
+import * as React from 'react';
```
Comment (IlyaVi) [inline app/components/baseComponents/SmRadioButtons/SmRadioButtons.js]: import

File: app/components/baseComponents/SmRadioButtons/SmRadioButtons.js [hunk 15]
```diff
@@ -0,0 +1,167 @@
...
+import { smFonts as Fonts, smColors as Colors } from '../../../vars';
```
Comment (IlyaVi) [inline app/components/baseComponents/SmRadioButtons/SmRadioButtons.js]: don't rename

File: app/components/baseComponents/SmRadioButtons/SmRadioButtons.js [hunk 16]
```diff
@@ -0,0 +1,167 @@
...
+          backgroundColor: Colors.white,
+          cursor: 'default'
+        };
+      }
+      return { borderColor: Colors.borderGray, backgroundColor: Colors.white };
+    };
+
+    const textLabelDisabledStyle = (val: RadioEntry) => (disabled || val.disabled ? styles.disabledLabelText : {});
+
+    const radioButtonElem = (val: RadioEntry) => (
```
Comment (IlyaVi) [inline app/components/baseComponents/SmRadioButtons/SmRadioButtons.js]: functions that return react element should start with "render.." and be class function

File: app/vars/fonts.js (hunk 9 above)
Comment (yhaspel) [reply app/vars/fonts.js]: done.

File: app/components/baseComponents/SendReceiveButton/SendReceiveButton.js (hunk 10 above)
Comment (yhaspel) [reply app/components/baseComponents/SendReceiveButton/SendReceiveButton.js]: done

File: app/components/baseComponents/SendReceiveButton/SendReceiveButton.js (hunk 11 above)
Comment (yhaspel) [reply app/components/baseComponents/SendReceiveButton/SendReceiveButton.js]: done

File: app/components/baseComponents/SendReceiveButton/SendReceiveButton.js (hunk 12 above)
Comment (yhaspel) [reply app/components/baseComponents/SendReceiveButton/SendReceiveButton.js]: done

File: app/components/baseComponents/SendReceiveButton/SendReceiveButton.js (hunk 13 above)
Comment (yhaspel) [reply app/components/baseComponents/SendReceiveButton/SendReceiveButton.js]: done

File: app/components/baseComponents/SmRadioButtons/SmRadioButtons.js (hunk 14 above)
Comment (yhaspel) [reply app/components/baseComponents/SmRadioButtons/SmRadioButtons.js]: done

File: app/components/baseComponents/SmRadioButtons/SmRadioButtons.js (hunk 15 above)
Comment (yhaspel) [reply app/components/baseComponents/SmRadioButtons/SmRadioButtons.js]: done

**Pull request merged**